master/lustregittagpoller.py

```diff
@@ -0,0 +1,224 @@
+# -*- python -*-
+# ex: set syntax=python:
+
+import itertools
+import os
+import re
+import urllib
+
+from twisted.internet import defer
+from twisted.internet import utils
+from twisted.python import log
+
+from buildbot import config
+from buildbot.changes import base 
+from buildbot.util import epoch2datetime
+from buildbot.util.state import StateMixin
+
+class LustreTagPoller(base.PollingChangeSource, StateMixin):
+
+    """ This source will poll a remote git repo for
+    new tags and will submit them to the change master."""
+
+    def __init__(self, repourl, workdir=None,
+                 pollInterval= 10 * 60, gitbin='git',
+                 category=None, project=None,
+                 pollinterval=-2, pollAtLaunch=False,
+                 encoding='utf-8', usetimestamps=True):
+
+        if pollinterval != -2:
```

Comment:
I have absolutely no idea what you are trying to accomplish here.  Neither will anyone else, because there are no comments explaining this.  But even without knowing what is going on, I can tell you that it is not at all wise to have two options that are named nearly identically, differing only in the capitalization of a single letter in the middle.